apollo-api/src/main/java/com/apollographql/apollo/api/Field.java

```diff
@@ -254,6 +382,9 @@ public ScalarType scalarType() {
     }
   }
 
+  /**
+   * Abstraction for a Field representing a conditional type.
+   */
   public static final class ConditionalTypeField extends Field {
```

Comment:
Conditional Type used for parsing inline fragments or fragments. Here is an example how it used:
```
final Field[] fields = {
        Field.forConditionalType("__typename", "__typename", new Field.ConditionalTypeReader<Fragments>() {
          @Override
          public Fragments read(String conditionalType, ResponseReader reader) throws IOException {
            return fragmentsFieldMapper.map(reader, conditionalType);
          }
        })
      };
```
It means that first field `__typename` will be read and then passed  to another nested mapper along with `reader`  that will decide by checking `conditionalType` what type of fragment it will parse.